### PR TITLE
[Improve] Flink-Yarn multi-version adaptation

### PR DESCRIPTION
**Pull request by lizh501**
## Brief change log

Flink-Yarn multi-version adaptation. It follows the design pattern of streampark-flink-shims. Make it more flexible for flink on yarn job submission code dev in multiple versions of flink-yarn. It also fixes the problem of Yarn per-job submission failure in recent versions of Flink (Flink 1.19+).

## Verifying this change

This change is already covered by existing tests

## Does this pull request potentially affect one of the following parts

Dependencies (does it add or upgrade a dependency): no

### CODE REVIEW

File: streampark-flink/streampark-flink-shims/streampark-flink-shims-base/pom.xml [hunk 1]
```diff
@@ -105,6 +105,19 @@
             <artifactId>hadoop-client-runtime</artifactId>
             <optional>true</optional>
         </dependency>
+
+        <dependency>
+            <groupId>org.apache.flink</groupId>
+            <artifactId>flink-yarn${scala.binary.flink.version}</artifactId>
+            <version>${flink.version}</version>
+            <scope>provided</scope>
+            <exclusions>
```
Comment (wolfboys) [inline streampark-flink/streampark-flink-shims/streampark-flink-shims-base/pom.xml]: Maven scope is "provided". We don't need to specify exclusions. These dependencies will be automatically excluded.

Comment (lizh501) [reply streampark-flink/streampark-flink-shims/streampark-flink-shims-base/pom.xml]: OK, I understand this. Thank you. And it is fixed.

File: streampark-flink/streampark-flink-shims/streampark-flink-shims_flink-1.20/src/main/scala/org/apache/streampark/flink/core/YarnClusterDescriptorWrapper.scala [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+  override def addShipFiles(input: util.List[File]) = {
+    val shipFiles: Seq[File] = input.asScala
+    var paths: ArrayBuffer[Path] = ArrayBuffer[Path]()
+    for (item <- shipFiles) {
+      val path = new Path(item.toURI)
+      paths += path
+    }
+    yarnClusterDescriptor.addShipFiles(paths.asJava)
+  }
```
Comment (wolfboys) [inline streampark-flink/streampark-flink-shims/streampark-flink-shims_flink-1.20/src/main/scala/org/apache/streampark/flink/core/YarnClusterDescriptorWrapper.scala:34]: We can make further improvements to the code
```
def addShipFiles(input: util.List[File]) = {
    val f = input.map(c => new Path(c.toURI)).asJava
    yarnClusterDescriptor.addShipFiles(f)
 }
```

Comment (lizh501) [reply streampark-flink/streampark-flink-shims/streampark-flink-shims_flink-1.20/src/main/scala/org/apache/streampark/flink/core/YarnClusterDescriptorWrapper.scala:34]: Thank you for your advise. I do this improvement now.

**Review by wolfboys — APPROVED** [review]
LGTM

**Pull request merged**